src/Client.php

```diff
@@ -99,39 +135,66 @@ public function register(string $identifier, string $password, array $attributes
      */
     public function changePassword(string $identifier, string $newPassword): bool
     {
-        // TODO: Implement changePassword() method.
+        return $this->changeAttributes($identifier, ['userPassword' => $this->generatePassword($newPassword)]);
     }
 
     /**
      * @inheritDoc
      */
     public function changeAttribute(string $identifier, string $key, string $value): bool
     {
-        // TODO: Implement changeAttribute() method.
+        return $this->changeAttributes($identifier, [$key => $value]);
     }
 
     /**
      * @inheritDoc
      */
     public function changeAttributes(string $identifier, array $attributes): bool
     {
-        // TODO: Implement changeAttributes() method.
+        $formattedAttributes = $this->formatAttributes($attributes);
+
+        $operations = [];
+
+        foreach ($formattedAttributes as $key => $value) {
+            $operations[$key] = new UpdateOperation(LDAP_MODIFY_BATCH_REPLACE, $key, Arr::wrap($value));
+        }
+
+        try {
+            /**
+             * @var EntryManager $entryManager
+             */
+            $entryManager = $this->ldap->getEntryManager();
+
+            $dn = $this->buildDn($identifier);
+
+            $entry = $this->getLdapEntry($dn);
+
+            $entryManager->applyOperations($entry->getDn(), $operations);
+        } catch (ExceptionInterface $e) {
+            throw new ClientException($e->getMessage(), (int) $e->getCode(), $e);
+        }
+
+        return true;
     }
 
     /**
      * @inheritDoc
      */
     public function enableUser(string $identifier): bool
     {
-        // TODO: Implement enableUser() method.
+        $this->changeAttribute($identifier, 'userAccountControl', '0');
+
+        return true;
     }
 
     /**
      * @inheritDoc
      */
     public function disableUser(string $identifier): bool
     {
-        // TODO: Implement disableUser() method.
+        $this->changeAttribute($identifier, 'userAccountControl', '2');
```

Comment:
So if `2` is disabled what does 1 mean?

Reply:
It was my poor attempt to follow the spec for `userAccountControl` field - In Microsoft the field is `0x0002` for disabled accounts, and `0x0200` for enabled accounts - (but for some reason I later reduced this to `1` for enabled and `2` for disabled).

Do you have a recommendation to deal with this?

Reply:
Just pushed update to move these values to constants instead, which will also reflect the MS spec for account control. Hopefully an improvement.